src/OpenLoco/src/Vehicles/Vehicle.cpp

```diff
@@ -172,11 +173,87 @@ namespace OpenLoco::Vehicles
     }
 
     // 0x004AA464
-    void VehicleBase::sub_4AA464()
+    void VehicleBase::destroyTrain()
     {
-        registers regs;
-        regs.esi = X86Pointer(this);
-        call(0x004AA464, regs);
+        VehicleHead* head = EntityManager::get<VehicleHead>(this->getHead());
```

Comment:
create a `Vehicle train(getHead());` then you have access to all the components safely.